src/fivetran-destination/src/destination/dml.rs

```diff
@@ -388,6 +407,152 @@ async fn update_file(
+        .context("time went backwards!")?;
+    // TODO(parkmycar): Make sure table name is <= 255 characters.
+    let temp_table_name = format!("fivetran_{}_temp_delete_{}", table.name, ts.as_millis());
+
+    let schema_name = escape::escape_identifier(schema);
+    let table_name = escape::escape_identifier(&table.name);
+    let temp_table_name = escape::escape_identifier(&temp_table_name);
+
+    let columns = table
+        .columns
+        .iter()
+        .map(|col| {
+            let mut ty: Cow<'static, str> = utils::to_materialize_type(col.r#type())?.into();
+            if let Some(d) = &col.decimal {
+                ty.to_mut()
+                    .push_str(&format!("({}, {})", d.precision, d.scale));
+            }
+
+            Ok(ColumnMetadata {
+                name: col.name.to_string(),
+                ty,
+                is_primary: col.primary_key || col.name.to_lowercase() == FIVETRAN_SYSTEM_COLUMN_ID,
+            })
+        })
+        .collect::<Result<Vec<_>, anyhow::Error>>()?;
+
+    // Create our temporary table that we'll copy the delete files into.
+    let defs = columns
+        .iter()
+        .map(|col| format!("{} {}", col.name, col.ty))
+        .join(",");
+    let create_table_stmt = format!("CREATE TABLE {schema_name}.{temp_table_name} ({defs})");
+    client.batch_execute(&create_table_stmt).await?;
+
+    // TODO(parkmycar): Ideally we'd use the COPY FROM protocol here, but we need to add support
+    // for compression and encryption.
+    let cols = columns
+        .iter()
+        .enumerate()
+        .map(|(idx, col)| format!("${}::{}", idx + 1, col.ty))
+        .join(",");
+    let insert_stmt = format!("INSERT INTO {schema_name}.{temp_table_name} VALUES ({cols})",);
+    let insert_stmt = client
+        .prepare(&insert_stmt)
+        .await
+        .context("internal error: preparing delete temp table insert statement")?;
+
+    let mut total_count = 0;
+    for path in delete_files {
+        let reader = load_file(file_config, path)
+            .await
+            .with_context(|| format!("loading delete file {path}"))?;
+        let copied = copy_delete_file(file_config, client, &insert_stmt, reader)
+            .await
+            .with_context(|| format!("handling delete file {path}"))?;
+        total_count += copied;
+    }
+    tracing::info!(total_count, "inserted rows to {schema}.{temp_table_name}");
+
+    // Skip the update if there are no rows to delete!
+    if total_count > 0 {
+        // Mark rows as deleted by "merging" our temporary table into the destination table.
+        //
+        // TODO(parkmycar): Fivetran provides a column name in truncate for soft deletes, should the
+        // delete operation be doing the same?
+
+        // HACKY: We want to update the "_fivetran_synced" column for all of the rows we marked as
+        // deleted, but don't have a way to read from the temp table that would allow this in an
```

Comment:
I don't understand "but don't have a way to read from the temp table" if the temp table is referenced in the WHERE clause below. That seems to me to be reading from the temp table.

Reply:
The core issue is AFAICT there is no way to update a column in `t1` with a value from `t1_temp`. What I want to do is something like this:
```
UPDATE t1
SET _synced = t1_temp._synced
FROM t1_temp
WHERE t1.a = t1_temp.a
  AND t1.b = t1_temp.b
```
But we don't support the `FROM` clause in our `UPDATE` statements. AFAICT the only place we can refer to `t1_temp` in MZ is a subquery in the `WHERE` clause, and the scope of this subquery does not exist to a `SET` clause.

Reply:
We should support `FROM`! It's not too much different than `INSERT SELECT` which we already support. Mostly parsing and plumbing additions, but all the machinery exists to support it. Also see https://github.com/MaterializeInc/materialize/issues/9274 which is similarish.

Reply:
I agree! Do you mind if I do that in a follow up? We have a goal to give Fivetran a binary by EOW so they can start testing things on their end

Reply:
Yes, followup sounds correct.